src/CostModelDynamicLevel.sol

```diff
@@ -230,19 +230,21 @@ contract CostModelDynamicLevel is ICostModel {
   /// @param timeDelta_ Time since last update.
   function _computeNewCostFactorInOptimalZone(uint256 utilization_, uint256 timeDelta_) internal view returns (uint256) {
     uint256 currentCostFactorInOptimalZone_ = costFactorInOptimalZone;
-    if (utilization_ >= uOpt) {
-      // Cost factor increases with `timeDelta` and `utilization - uOpt`, but with a ceiling set at
-      // `costFactorAtFullUtilization`.
+    if (timeDelta_ == 0 || utilization_ == uOpt) return currentCostFactorInOptimalZone_;
+    if (utilization_ > uOpt) {
+      uint256 deviation_ = timeDelta_.mulDivUp(utilization_ - uOpt, FixedPointMathLib.WAD);
       return _min(
-        currentCostFactorInOptimalZone_ + optimalZoneRate.mulWadUp((utilization_ - uOpt) * timeDelta_),
-        costFactorAtFullUtilization
+        costFactorAtFullUtilization,
+        currentCostFactorInOptimalZone_.mulWadUp(FixedPointMathLib.WAD + optimalZoneRate * deviation_)
       );
     } else {
-      // Cost factor decreases with `timeDelta` and `utilization - uOpt`, but with a floor set at
-      // `costFactorAtZeroUtilization`.
-      uint256 delta_ = optimalZoneRate.mulWadUp((uOpt - utilization_) * timeDelta_);
-      if (delta_ > currentCostFactorInOptimalZone_ - costFactorAtZeroUtilization) return costFactorAtZeroUtilization;
-      else return currentCostFactorInOptimalZone_ - delta_;
+      uint256 deviation_ = timeDelta_.mulDivUp(uOpt - utilization_, FixedPointMathLib.WAD);
```

Comment:
Maybe better to do the div by wad on the next line for precision?

Reply:
Makes sense -- Changed, so I multiply everything together first and then scale by WAD^2.

74e479a8887741792f407c6d1fc6fe0814ffaa01

Reply:
<img width="895" alt="Screen Shot 2023-09-11 at 11 42 57 AM" src="https://github.com/Cozy-Finance/cozy-models-v2/assets/22300733/7e104d65-9b7f-4690-9f2f-788df4801ebc">

I'm still confused why we need to do WAD ** 2 in the numerator of the new calc? 😅

Reply:
Ahh sorry -- so for the case where `utilization > uOpt`, we want to grow the cost factor as `currentCostFactor*(1 + growthFactor)`. Basically we need the WAD^2, so the underlying calculation is `currentCostFactor*(WAD^2 + growthFactor*WAD^2) / WAD^2`. 

To be clearer, I'll change this so it matches what we do in the `utilization < uOpt` case, where we just compute the delta and then add it.

Reply:
ohhhh I see. thanks for explaining

Reply:
Changed here: 1334ec044cba05685106225f06384f18a0e7101f